modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2795,26 +2761,28 @@ private CompletableFuture<CompletableFuture<?>> applyUpdateCommand(
                 return resultFuture.thenApply(res -> {
                     UpdateCommandResult updateCommandResult = (UpdateCommandResult) res;
 
-                    if (full && !updateCommandResult.isPrimaryReplicaMatch()) {
+                    if (full && updateCommandResult != null && !updateCommandResult.isPrimaryReplicaMatch()) {
```

Comment:
I think this is done to handle a test result (SKIP_REPLICATION_IN_BENCHMARK).
Maybe it would be better to give the correct response to avoid this checking.

Reply:
This is not possible due to dependency issues. The response class is not available in raft module.